{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Sample Code for deconvolution\n",
    "***"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import matplotlib.pyplot as plt\n",
    "\n",
    "from deconv.deconv import Deconvolution"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_mix=\"\"\n",
    "file_ref=\"\"\n",
    "file_annotation_ref=\"D:/Gdrive/datasource/ann_file/Rat_transcriptID_RGD.csv\"\n",
    "\n",
    "#df_mix=pd.read_csv(file_mix, index_col=0)\n",
    "#df_ref=pd.read_csv(file_ref, index_col=0)\n",
    "df_annotation_ref=pd.read_csv(file_annotation_ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv(\"D:/Gdrive/notebook/2004_Transcriptome_Deconvolution/220217_RNAseq_reanalysis/data/gene.txt\",index_col=0,delimiter=\"\\t\")\n",
    "df_mix = df.loc[:,(df.columns.str.contains(\"liver\")) | (df.columns.str.contains(\"wbc\")) | (df.columns.str.contains(\"spleen\"))]\n",
    "df_ref = df.drop(df_mix.columns.tolist(), axis=1)\n",
    "df_mix.columns = [f\"{i.split('_')[0]}_{i.split('_')[1]}\" if not 'liver' in i else f\"{i.split('_')[0]}{i.split('_')[1]}_{i.split('_')[2]}\" for i in df_mix.columns]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat = Deconvolution()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat.set_data(df_mix, df_ref)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "annotation\n",
      "trimming\n",
      "tranformation\n",
      "1 : quantile\n",
      "normalization\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\internals\\blocks.py:402: RuntimeWarning: divide by zero encountered in log2\n",
      "  result = func(self.values, **kwargs)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1 : raw\n"
     ]
    }
   ],
   "source": [
    "dat.preprocessing_mix(df_ref=df_annotation_ref, places=[0,1],\n",
    "                      trimming=True,batch=False,combat=False,\n",
    "                      trans_method=\"log2\",\n",
    "                      norm_method_list=[\"quantile\"])\n",
    "\n",
    "dat.preprocessing_ref(df_ref=df_annotation_ref, places=[0,1],\n",
    "                      trimming=False,batch=False,combat=False,\n",
    "                      trans_method=\"log2\",\n",
    "                      norm_method_list=[\"raw\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "method : ttest\n",
      "extracted genes number = 6042\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\numpy\\core\\fromnumeric.py:3438: FutureWarning: In a future version, DataFrame.mean(axis=None) will return a scalar mean over the entire DataFrame. To retain the old behavior, use 'frame.mean(axis=0)' or just 'frame.mean()'\n",
      "  return mean(axis=axis, dtype=dtype, out=out, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "dat.deg(method=\"ttest\",\n",
    "        sep=\"_\",number=150,limit_CV=1,limit_FC=1.,log2=False,q_limit=0.05,\n",
    "        plot=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dat.fit(method=\"elasticnet\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"None of [Index(['B', 'CD4T', 'CD8T', 'Mon', 'NK', 'Neu'], dtype='object')] are in the [index]\"",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[1;32mC:\\Users\\KATSUH~1\\AppData\\Local\\Temp/ipykernel_12948/2225487570.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdat\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mplot_res\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mdat\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget_data\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mcontrol_names\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"liverctrl\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32mg:\\共有ドライブ\\2018_MoritaK\\publication\\ratdeconv\\deconv\\deconv\\deconv.py\u001b[0m in \u001b[0;36mplot_res\u001b[1;34m(self, sort_index, control_names, row_n, col_n)\u001b[0m\n\u001b[0;32m     98\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     99\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mplot_res\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mlist\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcontrol_names\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mlist\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m\"control, ctrl\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mrow_n\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mint\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m2\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mcol_n\u001b[0m\u001b[1;33m:\u001b[0m\u001b[0mint\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;36m3\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 100\u001b[1;33m         \u001b[0mplotter\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mplot_immune_box\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__res\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mcontrol_names\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mcontrol_names\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mrow_n\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mrow_n\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mcol_n\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mcol_n\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    101\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    102\u001b[0m     \u001b[1;31m### in/out put control ###\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mg:\\共有ドライブ\\2018_MoritaK\\publication\\ratdeconv\\deconv\\deconv\\_utils\\plotter.py\u001b[0m in \u001b[0;36mplot_immune_box\u001b[1;34m(df, sort_index, control_names, row_n, col_n)\u001b[0m\n\u001b[0;32m     46\u001b[0m     \u001b[0mimmunes\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtolist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     47\u001b[0m     \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mi\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"_\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 48\u001b[1;33m     \u001b[0mdf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0msort_index\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     49\u001b[0m     \u001b[0mfig\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mplt\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfigure\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mfigsize\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;36m5\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0mcol_n\u001b[0m\u001b[1;33m,\u001b[0m\u001b[1;36m5\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0mrow_n\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     50\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[1;33m,\u001b[0m\u001b[0mimmune\u001b[0m \u001b[1;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mimmunes\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexing.py\u001b[0m in \u001b[0;36m__getitem__\u001b[1;34m(self, key)\u001b[0m\n\u001b[0;32m    965\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    966\u001b[0m             \u001b[0mmaybe_callable\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mcom\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mapply_if_callable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 967\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmaybe_callable\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    968\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    969\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_is_scalar_access\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mtuple\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexing.py\u001b[0m in \u001b[0;36m_getitem_axis\u001b[1;34m(self, key, axis)\u001b[0m\n\u001b[0;32m   1189\u001b[0m                     \u001b[1;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"Cannot index with multidimensional key\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1190\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1191\u001b[1;33m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_getitem_iterable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1192\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1193\u001b[0m             \u001b[1;31m# nested tuple slicing\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexing.py\u001b[0m in \u001b[0;36m_getitem_iterable\u001b[1;34m(self, key, axis)\u001b[0m\n\u001b[0;32m   1130\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1131\u001b[0m         \u001b[1;31m# A collection of keys\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1132\u001b[1;33m         \u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_listlike_indexer\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1133\u001b[0m         return self.obj._reindex_with_indexers(\n\u001b[0;32m   1134\u001b[0m             \u001b[1;33m{\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;33m[\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mallow_dups\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexing.py\u001b[0m in \u001b[0;36m_get_listlike_indexer\u001b[1;34m(self, key, axis)\u001b[0m\n\u001b[0;32m   1325\u001b[0m         \u001b[0maxis_name\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_axis_name\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1326\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 1327\u001b[1;33m         \u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0max\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_get_indexer_strict\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   1328\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   1329\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36m_get_indexer_strict\u001b[1;34m(self, key, axis_name)\u001b[0m\n\u001b[0;32m   5780\u001b[0m             \u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnew_indexer\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_reindex_non_unique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5781\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5782\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_raise_if_missing\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5783\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5784\u001b[0m         \u001b[0mkeyarr\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mtake\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mC:\\ProgramData\\miniconda3\\envs\\deconv\\lib\\site-packages\\pandas\\core\\indexes\\base.py\u001b[0m in \u001b[0;36m_raise_if_missing\u001b[1;34m(self, key, indexer, axis_name)\u001b[0m\n\u001b[0;32m   5840\u001b[0m                 \u001b[1;32mif\u001b[0m \u001b[0muse_interval_msg\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5841\u001b[0m                     \u001b[0mkey\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5842\u001b[1;33m                 \u001b[1;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34mf\"None of [{key}] are in the [{axis_name}]\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5843\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5844\u001b[0m             \u001b[0mnot_found\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mensure_index\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mmissing_mask\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mnonzero\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mKeyError\u001b[0m: \"None of [Index(['B', 'CD4T', 'CD8T', 'Mon', 'NK', 'Neu'], dtype='object')] are in the [index]\""
     ]
    }
   ],
   "source": [
    "dat.plot_res(sort_index=dat.get_data()[2].columns.tolist(),control_names=\"liverctrl\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>liver_apap_1</th>\n",
       "      <th>liver_apap_2</th>\n",
       "      <th>liver_apap_3</th>\n",
       "      <th>liver_apap_4</th>\n",
       "      <th>liver_apap_5</th>\n",
       "      <th>liver_ctrl_1</th>\n",
       "      <th>liver_ctrl_2</th>\n",
       "      <th>liver_ctrl_3</th>\n",
       "      <th>liver_ctrl_4</th>\n",
       "      <th>liver_ctrl_5</th>\n",
       "      <th>...</th>\n",
       "      <th>spleen_1</th>\n",
       "      <th>spleen_2</th>\n",
       "      <th>spleen_3</th>\n",
       "      <th>spleen_4</th>\n",
       "      <th>spleen_5</th>\n",
       "      <th>wbc_1</th>\n",
       "      <th>wbc_2</th>\n",
       "      <th>wbc_3</th>\n",
       "      <th>wbc_4</th>\n",
       "      <th>wbc_5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>4.171892</td>\n",
       "      <td>4.368763</td>\n",
       "      <td>2.679840</td>\n",
       "      <td>3.457114</td>\n",
       "      <td>3.352063</td>\n",
       "      <td>3.051128</td>\n",
       "      <td>3.692861</td>\n",
       "      <td>3.046806</td>\n",
       "      <td>4.147998</td>\n",
       "      <td>4.088962</td>\n",
       "      <td>...</td>\n",
       "      <td>0.687839</td>\n",
       "      <td>2.738042</td>\n",
       "      <td>2.142524</td>\n",
       "      <td>1.767811</td>\n",
       "      <td>1.708707</td>\n",
       "      <td>2.063018</td>\n",
       "      <td>2.317794</td>\n",
       "      <td>3.161509</td>\n",
       "      <td>2.396825</td>\n",
       "      <td>3.121917</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>4.777583</td>\n",
       "      <td>5.135787</td>\n",
       "      <td>4.406078</td>\n",
       "      <td>5.164614</td>\n",
       "      <td>4.820383</td>\n",
       "      <td>4.579940</td>\n",
       "      <td>2.922225</td>\n",
       "      <td>4.356737</td>\n",
       "      <td>4.302464</td>\n",
       "      <td>2.955057</td>\n",
       "      <td>...</td>\n",
       "      <td>5.358663</td>\n",
       "      <td>5.301582</td>\n",
       "      <td>4.990571</td>\n",
       "      <td>5.275613</td>\n",
       "      <td>5.340321</td>\n",
       "      <td>4.177978</td>\n",
       "      <td>4.500131</td>\n",
       "      <td>4.859896</td>\n",
       "      <td>4.592598</td>\n",
       "      <td>4.904986</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAAS</th>\n",
       "      <td>6.296436</td>\n",
       "      <td>6.340145</td>\n",
       "      <td>6.170978</td>\n",
       "      <td>6.256932</td>\n",
       "      <td>6.196240</td>\n",
       "      <td>5.989640</td>\n",
       "      <td>5.876509</td>\n",
       "      <td>5.987445</td>\n",
       "      <td>5.916353</td>\n",
       "      <td>5.877821</td>\n",
       "      <td>...</td>\n",
       "      <td>6.104736</td>\n",
       "      <td>6.114602</td>\n",
       "      <td>6.102993</td>\n",
       "      <td>6.115497</td>\n",
       "      <td>6.031997</td>\n",
       "      <td>5.894273</td>\n",
       "      <td>5.799512</td>\n",
       "      <td>5.682490</td>\n",
       "      <td>5.892925</td>\n",
       "      <td>6.002100</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AACS</th>\n",
       "      <td>5.084872</td>\n",
       "      <td>5.340559</td>\n",
       "      <td>5.179738</td>\n",
       "      <td>4.780748</td>\n",
       "      <td>4.892280</td>\n",
       "      <td>5.898287</td>\n",
       "      <td>5.492364</td>\n",
       "      <td>5.522481</td>\n",
       "      <td>5.582882</td>\n",
       "      <td>5.536895</td>\n",
       "      <td>...</td>\n",
       "      <td>6.277818</td>\n",
       "      <td>6.294675</td>\n",
       "      <td>6.189399</td>\n",
       "      <td>6.302376</td>\n",
       "      <td>6.280773</td>\n",
       "      <td>6.378833</td>\n",
       "      <td>6.343202</td>\n",
       "      <td>6.436363</td>\n",
       "      <td>6.334977</td>\n",
       "      <td>6.341985</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AAGAB</th>\n",
       "      <td>6.100605</td>\n",
       "      <td>6.268855</td>\n",
       "      <td>6.080500</td>\n",
       "      <td>5.974795</td>\n",
       "      <td>6.046206</td>\n",
       "      <td>6.077183</td>\n",
       "      <td>6.050617</td>\n",
       "      <td>6.014483</td>\n",
       "      <td>5.863288</td>\n",
       "      <td>6.029056</td>\n",
       "      <td>...</td>\n",
       "      <td>6.559908</td>\n",
       "      <td>6.500956</td>\n",
       "      <td>6.467008</td>\n",
       "      <td>6.508072</td>\n",
       "      <td>6.494771</td>\n",
       "      <td>6.286339</td>\n",
       "      <td>6.250032</td>\n",
       "      <td>6.259199</td>\n",
       "      <td>6.172023</td>\n",
       "      <td>6.348596</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ZZZ3</th>\n",
       "      <td>5.059501</td>\n",
       "      <td>5.089095</td>\n",
       "      <td>5.379766</td>\n",
       "      <td>5.129119</td>\n",
       "      <td>5.199992</td>\n",
       "      <td>5.057159</td>\n",
       "      <td>4.907567</td>\n",
       "      <td>4.880658</td>\n",
       "      <td>5.111144</td>\n",
       "      <td>5.095581</td>\n",
       "      <td>...</td>\n",
       "      <td>5.642742</td>\n",
       "      <td>5.567245</td>\n",
       "      <td>5.584156</td>\n",
       "      <td>5.576639</td>\n",
       "      <td>5.610687</td>\n",
       "      <td>5.697179</td>\n",
       "      <td>5.600893</td>\n",
       "      <td>5.650504</td>\n",
       "      <td>5.655223</td>\n",
       "      <td>5.619316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL11</th>\n",
       "      <td>6.661719</td>\n",
       "      <td>6.517437</td>\n",
       "      <td>6.557653</td>\n",
       "      <td>6.536132</td>\n",
       "      <td>6.539822</td>\n",
       "      <td>6.622350</td>\n",
       "      <td>6.611776</td>\n",
       "      <td>6.615163</td>\n",
       "      <td>6.561614</td>\n",
       "      <td>6.573712</td>\n",
       "      <td>...</td>\n",
       "      <td>6.151907</td>\n",
       "      <td>6.081989</td>\n",
       "      <td>6.098274</td>\n",
       "      <td>6.097971</td>\n",
       "      <td>6.114829</td>\n",
       "      <td>5.910433</td>\n",
       "      <td>5.983458</td>\n",
       "      <td>5.966022</td>\n",
       "      <td>6.006913</td>\n",
       "      <td>6.090774</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL24</th>\n",
       "      <td>6.635006</td>\n",
       "      <td>6.663041</td>\n",
       "      <td>6.608876</td>\n",
       "      <td>6.641167</td>\n",
       "      <td>6.572851</td>\n",
       "      <td>6.548315</td>\n",
       "      <td>6.461192</td>\n",
       "      <td>6.511042</td>\n",
       "      <td>6.434946</td>\n",
       "      <td>6.351644</td>\n",
       "      <td>...</td>\n",
       "      <td>6.378661</td>\n",
       "      <td>6.346775</td>\n",
       "      <td>6.319169</td>\n",
       "      <td>6.367975</td>\n",
       "      <td>6.378208</td>\n",
       "      <td>6.330969</td>\n",
       "      <td>6.304709</td>\n",
       "      <td>6.234951</td>\n",
       "      <td>6.333053</td>\n",
       "      <td>6.326023</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL9</th>\n",
       "      <td>6.403059</td>\n",
       "      <td>6.467155</td>\n",
       "      <td>6.405453</td>\n",
       "      <td>6.418367</td>\n",
       "      <td>6.382982</td>\n",
       "      <td>6.308071</td>\n",
       "      <td>6.250743</td>\n",
       "      <td>6.390397</td>\n",
       "      <td>6.448732</td>\n",
       "      <td>6.398578</td>\n",
       "      <td>...</td>\n",
       "      <td>6.653467</td>\n",
       "      <td>6.607990</td>\n",
       "      <td>6.622350</td>\n",
       "      <td>6.636533</td>\n",
       "      <td>6.653027</td>\n",
       "      <td>6.562646</td>\n",
       "      <td>6.524847</td>\n",
       "      <td>6.505730</td>\n",
       "      <td>6.575887</td>\n",
       "      <td>6.590958</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNF141</th>\n",
       "      <td>6.591483</td>\n",
       "      <td>6.536542</td>\n",
       "      <td>6.378922</td>\n",
       "      <td>6.602274</td>\n",
       "      <td>6.607857</td>\n",
       "      <td>6.740968</td>\n",
       "      <td>6.846775</td>\n",
       "      <td>6.808243</td>\n",
       "      <td>6.883525</td>\n",
       "      <td>6.752931</td>\n",
       "      <td>...</td>\n",
       "      <td>6.257703</td>\n",
       "      <td>6.318356</td>\n",
       "      <td>6.091488</td>\n",
       "      <td>6.194379</td>\n",
       "      <td>6.168598</td>\n",
       "      <td>6.279508</td>\n",
       "      <td>6.297505</td>\n",
       "      <td>6.355925</td>\n",
       "      <td>6.305430</td>\n",
       "      <td>6.232299</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10153 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         liver_apap_1  liver_apap_2  liver_apap_3  liver_apap_4  liver_apap_5  \\\n",
       "A2ML1        4.171892      4.368763      2.679840      3.457114      3.352063   \n",
       "A3GALT2      4.777583      5.135787      4.406078      5.164614      4.820383   \n",
       "AAAS         6.296436      6.340145      6.170978      6.256932      6.196240   \n",
       "AACS         5.084872      5.340559      5.179738      4.780748      4.892280   \n",
       "AAGAB        6.100605      6.268855      6.080500      5.974795      6.046206   \n",
       "...               ...           ...           ...           ...           ...   \n",
       "ZZZ3         5.059501      5.089095      5.379766      5.129119      5.199992   \n",
       "MRPL11       6.661719      6.517437      6.557653      6.536132      6.539822   \n",
       "MRPL24       6.635006      6.663041      6.608876      6.641167      6.572851   \n",
       "MRPL9        6.403059      6.467155      6.405453      6.418367      6.382982   \n",
       "RNF141       6.591483      6.536542      6.378922      6.602274      6.607857   \n",
       "\n",
       "         liver_ctrl_1  liver_ctrl_2  liver_ctrl_3  liver_ctrl_4  liver_ctrl_5  \\\n",
       "A2ML1        3.051128      3.692861      3.046806      4.147998      4.088962   \n",
       "A3GALT2      4.579940      2.922225      4.356737      4.302464      2.955057   \n",
       "AAAS         5.989640      5.876509      5.987445      5.916353      5.877821   \n",
       "AACS         5.898287      5.492364      5.522481      5.582882      5.536895   \n",
       "AAGAB        6.077183      6.050617      6.014483      5.863288      6.029056   \n",
       "...               ...           ...           ...           ...           ...   \n",
       "ZZZ3         5.057159      4.907567      4.880658      5.111144      5.095581   \n",
       "MRPL11       6.622350      6.611776      6.615163      6.561614      6.573712   \n",
       "MRPL24       6.548315      6.461192      6.511042      6.434946      6.351644   \n",
       "MRPL9        6.308071      6.250743      6.390397      6.448732      6.398578   \n",
       "RNF141       6.740968      6.846775      6.808243      6.883525      6.752931   \n",
       "\n",
       "         ...  spleen_1  spleen_2  spleen_3  spleen_4  spleen_5     wbc_1  \\\n",
       "A2ML1    ...  0.687839  2.738042  2.142524  1.767811  1.708707  2.063018   \n",
       "A3GALT2  ...  5.358663  5.301582  4.990571  5.275613  5.340321  4.177978   \n",
       "AAAS     ...  6.104736  6.114602  6.102993  6.115497  6.031997  5.894273   \n",
       "AACS     ...  6.277818  6.294675  6.189399  6.302376  6.280773  6.378833   \n",
       "AAGAB    ...  6.559908  6.500956  6.467008  6.508072  6.494771  6.286339   \n",
       "...      ...       ...       ...       ...       ...       ...       ...   \n",
       "ZZZ3     ...  5.642742  5.567245  5.584156  5.576639  5.610687  5.697179   \n",
       "MRPL11   ...  6.151907  6.081989  6.098274  6.097971  6.114829  5.910433   \n",
       "MRPL24   ...  6.378661  6.346775  6.319169  6.367975  6.378208  6.330969   \n",
       "MRPL9    ...  6.653467  6.607990  6.622350  6.636533  6.653027  6.562646   \n",
       "RNF141   ...  6.257703  6.318356  6.091488  6.194379  6.168598  6.279508   \n",
       "\n",
       "            wbc_2     wbc_3     wbc_4     wbc_5  \n",
       "A2ML1    2.317794  3.161509  2.396825  3.121917  \n",
       "A3GALT2  4.500131  4.859896  4.592598  4.904986  \n",
       "AAAS     5.799512  5.682490  5.892925  6.002100  \n",
       "AACS     6.343202  6.436363  6.334977  6.341985  \n",
       "AAGAB    6.250032  6.259199  6.172023  6.348596  \n",
       "...           ...       ...       ...       ...  \n",
       "ZZZ3     5.600893  5.650504  5.655223  5.619316  \n",
       "MRPL11   5.983458  5.966022  6.006913  6.090774  \n",
       "MRPL24   6.304709  6.234951  6.333053  6.326023  \n",
       "MRPL9    6.524847  6.505730  6.575887  6.590958  \n",
       "RNF141   6.297505  6.355925  6.305430  6.232299  \n",
       "\n",
       "[10153 rows x 25 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.get_data()[0] # mix data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Mon_5</th>\n",
       "      <th>B_1</th>\n",
       "      <th>B_2</th>\n",
       "      <th>B_3</th>\n",
       "      <th>B_4</th>\n",
       "      <th>B_5</th>\n",
       "      <th>B_6</th>\n",
       "      <th>CD4T_1</th>\n",
       "      <th>CD4T_2</th>\n",
       "      <th>CD4T_3</th>\n",
       "      <th>...</th>\n",
       "      <th>Neu_3</th>\n",
       "      <th>Neu_4</th>\n",
       "      <th>Neu_5</th>\n",
       "      <th>Neu_6</th>\n",
       "      <th>NK_1</th>\n",
       "      <th>NK_2</th>\n",
       "      <th>NK_3</th>\n",
       "      <th>NK_4</th>\n",
       "      <th>NK_5</th>\n",
       "      <th>NK_6</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A1BG</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A1CF</th>\n",
       "      <td>1.917041</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.536623</td>\n",
       "      <td>0.749090</td>\n",
       "      <td>0.289926</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.305709</td>\n",
       "      <td>...</td>\n",
       "      <td>0.343877</td>\n",
       "      <td>0.350853</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.845547</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.555396</td>\n",
       "      <td>0.297339</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2M</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>1.487224</td>\n",
       "      <td>1.537942</td>\n",
       "      <td>0.524679</td>\n",
       "      <td>0.707685</td>\n",
       "      <td>0.513098</td>\n",
       "      <td>1.537140</td>\n",
       "      <td>0.164129</td>\n",
       "      <td>0.169150</td>\n",
       "      <td>0.546472</td>\n",
       "      <td>...</td>\n",
       "      <td>0.490985</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.179374</td>\n",
       "      <td>0.310239</td>\n",
       "      <td>0.185210</td>\n",
       "      <td>1.068251</td>\n",
       "      <td>0.739119</td>\n",
       "      <td>1.087921</td>\n",
       "      <td>0.852808</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A2ML1</th>\n",
       "      <td>1.255521</td>\n",
       "      <td>1.786561</td>\n",
       "      <td>0.998844</td>\n",
       "      <td>0.917930</td>\n",
       "      <td>1.223184</td>\n",
       "      <td>1.232711</td>\n",
       "      <td>1.687522</td>\n",
       "      <td>0.790131</td>\n",
       "      <td>1.809739</td>\n",
       "      <td>1.521277</td>\n",
       "      <td>...</td>\n",
       "      <td>0.627395</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.341491</td>\n",
       "      <td>1.298917</td>\n",
       "      <td>3.419195</td>\n",
       "      <td>2.442018</td>\n",
       "      <td>1.399680</td>\n",
       "      <td>1.243419</td>\n",
       "      <td>1.733623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A3GALT2</th>\n",
       "      <td>5.557582</td>\n",
       "      <td>4.816717</td>\n",
       "      <td>5.585095</td>\n",
       "      <td>4.594804</td>\n",
       "      <td>4.694271</td>\n",
       "      <td>4.310881</td>\n",
       "      <td>4.629785</td>\n",
       "      <td>3.725194</td>\n",
       "      <td>4.499781</td>\n",
       "      <td>4.319475</td>\n",
       "      <td>...</td>\n",
       "      <td>3.704725</td>\n",
       "      <td>3.313323</td>\n",
       "      <td>1.730822</td>\n",
       "      <td>4.792776</td>\n",
       "      <td>4.216264</td>\n",
       "      <td>4.259992</td>\n",
       "      <td>5.425130</td>\n",
       "      <td>5.212667</td>\n",
       "      <td>5.539806</td>\n",
       "      <td>4.441381</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL11</th>\n",
       "      <td>7.583597</td>\n",
       "      <td>7.871143</td>\n",
       "      <td>7.749989</td>\n",
       "      <td>7.587909</td>\n",
       "      <td>7.658721</td>\n",
       "      <td>7.719543</td>\n",
       "      <td>7.922954</td>\n",
       "      <td>7.915317</td>\n",
       "      <td>7.506486</td>\n",
       "      <td>7.823560</td>\n",
       "      <td>...</td>\n",
       "      <td>6.535465</td>\n",
       "      <td>6.029931</td>\n",
       "      <td>4.979148</td>\n",
       "      <td>6.694329</td>\n",
       "      <td>7.780850</td>\n",
       "      <td>7.610058</td>\n",
       "      <td>8.182406</td>\n",
       "      <td>7.811371</td>\n",
       "      <td>7.895387</td>\n",
       "      <td>7.884946</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL24</th>\n",
       "      <td>8.103654</td>\n",
       "      <td>8.467446</td>\n",
       "      <td>8.776172</td>\n",
       "      <td>8.361930</td>\n",
       "      <td>8.051174</td>\n",
       "      <td>8.429524</td>\n",
       "      <td>8.462320</td>\n",
       "      <td>8.778465</td>\n",
       "      <td>8.814563</td>\n",
       "      <td>8.666892</td>\n",
       "      <td>...</td>\n",
       "      <td>7.214256</td>\n",
       "      <td>7.191661</td>\n",
       "      <td>5.891028</td>\n",
       "      <td>7.916443</td>\n",
       "      <td>8.453455</td>\n",
       "      <td>8.442615</td>\n",
       "      <td>8.679606</td>\n",
       "      <td>8.601010</td>\n",
       "      <td>8.509067</td>\n",
       "      <td>8.387032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MRPL9</th>\n",
       "      <td>8.878859</td>\n",
       "      <td>9.119133</td>\n",
       "      <td>9.188242</td>\n",
       "      <td>8.821991</td>\n",
       "      <td>8.643862</td>\n",
       "      <td>8.871707</td>\n",
       "      <td>9.115993</td>\n",
       "      <td>9.238755</td>\n",
       "      <td>8.859089</td>\n",
       "      <td>9.092360</td>\n",
       "      <td>...</td>\n",
       "      <td>8.161771</td>\n",
       "      <td>8.171186</td>\n",
       "      <td>6.592357</td>\n",
       "      <td>8.669987</td>\n",
       "      <td>8.903112</td>\n",
       "      <td>8.920920</td>\n",
       "      <td>9.286540</td>\n",
       "      <td>9.337274</td>\n",
       "      <td>9.143264</td>\n",
       "      <td>9.088537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>RNF141</th>\n",
       "      <td>9.396386</td>\n",
       "      <td>7.986987</td>\n",
       "      <td>8.114582</td>\n",
       "      <td>7.830445</td>\n",
       "      <td>7.636022</td>\n",
       "      <td>7.509279</td>\n",
       "      <td>7.848900</td>\n",
       "      <td>7.920426</td>\n",
       "      <td>7.428591</td>\n",
       "      <td>7.925638</td>\n",
       "      <td>...</td>\n",
       "      <td>8.243592</td>\n",
       "      <td>8.548072</td>\n",
       "      <td>6.941688</td>\n",
       "      <td>8.981032</td>\n",
       "      <td>7.932717</td>\n",
       "      <td>7.982940</td>\n",
       "      <td>7.275819</td>\n",
       "      <td>7.951991</td>\n",
       "      <td>8.043913</td>\n",
       "      <td>8.084261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TGAP1</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>...</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19436 rows × 36 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            Mon_5       B_1       B_2       B_3       B_4       B_5       B_6  \\\n",
       "A1BG     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "A1CF     1.917041  0.000000  0.000000  0.000000  0.536623  0.749090  0.289926   \n",
       "A2M      0.000000  1.487224  1.537942  0.524679  0.707685  0.513098  1.537140   \n",
       "A2ML1    1.255521  1.786561  0.998844  0.917930  1.223184  1.232711  1.687522   \n",
       "A3GALT2  5.557582  4.816717  5.585095  4.594804  4.694271  4.310881  4.629785   \n",
       "...           ...       ...       ...       ...       ...       ...       ...   \n",
       "MRPL11   7.583597  7.871143  7.749989  7.587909  7.658721  7.719543  7.922954   \n",
       "MRPL24   8.103654  8.467446  8.776172  8.361930  8.051174  8.429524  8.462320   \n",
       "MRPL9    8.878859  9.119133  9.188242  8.821991  8.643862  8.871707  9.115993   \n",
       "RNF141   9.396386  7.986987  8.114582  7.830445  7.636022  7.509279  7.848900   \n",
       "TGAP1    0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000   \n",
       "\n",
       "           CD4T_1    CD4T_2    CD4T_3  ...     Neu_3     Neu_4     Neu_5  \\\n",
       "A1BG     0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "A1CF     0.000000  0.000000  0.305709  ...  0.343877  0.350853  0.000000   \n",
       "A2M      0.164129  0.169150  0.546472  ...  0.490985  0.000000  0.000000   \n",
       "A2ML1    0.790131  1.809739  1.521277  ...  0.627395  0.000000  0.000000   \n",
       "A3GALT2  3.725194  4.499781  4.319475  ...  3.704725  3.313323  1.730822   \n",
       "...           ...       ...       ...  ...       ...       ...       ...   \n",
       "MRPL11   7.915317  7.506486  7.823560  ...  6.535465  6.029931  4.979148   \n",
       "MRPL24   8.778465  8.814563  8.666892  ...  7.214256  7.191661  5.891028   \n",
       "MRPL9    9.238755  8.859089  9.092360  ...  8.161771  8.171186  6.592357   \n",
       "RNF141   7.920426  7.428591  7.925638  ...  8.243592  8.548072  6.941688   \n",
       "TGAP1    0.000000  0.000000  0.000000  ...  0.000000  0.000000  0.000000   \n",
       "\n",
       "            Neu_6      NK_1      NK_2      NK_3      NK_4      NK_5      NK_6  \n",
       "A1BG     0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "A1CF     0.845547  0.000000  0.000000  0.000000  0.555396  0.297339  0.000000  \n",
       "A2M      0.179374  0.310239  0.185210  1.068251  0.739119  1.087921  0.852808  \n",
       "A2ML1    0.341491  1.298917  3.419195  2.442018  1.399680  1.243419  1.733623  \n",
       "A3GALT2  4.792776  4.216264  4.259992  5.425130  5.212667  5.539806  4.441381  \n",
       "...           ...       ...       ...       ...       ...       ...       ...  \n",
       "MRPL11   6.694329  7.780850  7.610058  8.182406  7.811371  7.895387  7.884946  \n",
       "MRPL24   7.916443  8.453455  8.442615  8.679606  8.601010  8.509067  8.387032  \n",
       "MRPL9    8.669987  8.903112  8.920920  9.286540  9.337274  9.143264  9.088537  \n",
       "RNF141   8.981032  7.932717  7.982940  7.275819  7.951991  8.043913  8.084261  \n",
       "TGAP1    0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  0.000000  \n",
       "\n",
       "[19436 rows x 36 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.get_data()[1] # ref data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>B</th>\n",
       "      <th>CD4T</th>\n",
       "      <th>CD8T</th>\n",
       "      <th>Mon</th>\n",
       "      <th>NK</th>\n",
       "      <th>Neu</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>UBE2Q2L</th>\n",
       "      <td>4.067965</td>\n",
       "      <td>3.937985</td>\n",
       "      <td>3.688142</td>\n",
       "      <td>2.496551</td>\n",
       "      <td>6.377228</td>\n",
       "      <td>4.420924</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DNASE2B</th>\n",
       "      <td>3.131231</td>\n",
       "      <td>2.395820</td>\n",
       "      <td>2.648317</td>\n",
       "      <td>4.778933</td>\n",
       "      <td>3.695496</td>\n",
       "      <td>7.830362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BTG3</th>\n",
       "      <td>7.024362</td>\n",
       "      <td>5.826770</td>\n",
       "      <td>5.921869</td>\n",
       "      <td>5.217145</td>\n",
       "      <td>6.295500</td>\n",
       "      <td>4.514722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CASTOR1</th>\n",
       "      <td>5.101510</td>\n",
       "      <td>5.766280</td>\n",
       "      <td>4.211267</td>\n",
       "      <td>4.774645</td>\n",
       "      <td>4.472333</td>\n",
       "      <td>4.526286</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PLEK2</th>\n",
       "      <td>4.268419</td>\n",
       "      <td>3.486914</td>\n",
       "      <td>3.293703</td>\n",
       "      <td>0.537386</td>\n",
       "      <td>3.706866</td>\n",
       "      <td>0.652470</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TNFSF13</th>\n",
       "      <td>6.748283</td>\n",
       "      <td>6.212415</td>\n",
       "      <td>6.257585</td>\n",
       "      <td>10.367265</td>\n",
       "      <td>6.734800</td>\n",
       "      <td>9.328266</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AKAP2</th>\n",
       "      <td>7.195798</td>\n",
       "      <td>6.236059</td>\n",
       "      <td>6.895361</td>\n",
       "      <td>4.273324</td>\n",
       "      <td>6.506765</td>\n",
       "      <td>3.872548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GZMM</th>\n",
       "      <td>11.561187</td>\n",
       "      <td>10.916939</td>\n",
       "      <td>12.031797</td>\n",
       "      <td>9.859231</td>\n",
       "      <td>14.340722</td>\n",
       "      <td>10.027372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PROSER2</th>\n",
       "      <td>2.791149</td>\n",
       "      <td>1.533664</td>\n",
       "      <td>1.805223</td>\n",
       "      <td>1.360205</td>\n",
       "      <td>3.899127</td>\n",
       "      <td>1.952887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SLC41A3</th>\n",
       "      <td>4.788027</td>\n",
       "      <td>5.854505</td>\n",
       "      <td>5.468683</td>\n",
       "      <td>3.793310</td>\n",
       "      <td>4.700652</td>\n",
       "      <td>4.015085</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>773 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 B       CD4T       CD8T        Mon         NK        Neu\n",
       "UBE2Q2L   4.067965   3.937985   3.688142   2.496551   6.377228   4.420924\n",
       "DNASE2B   3.131231   2.395820   2.648317   4.778933   3.695496   7.830362\n",
       "BTG3      7.024362   5.826770   5.921869   5.217145   6.295500   4.514722\n",
       "CASTOR1   5.101510   5.766280   4.211267   4.774645   4.472333   4.526286\n",
       "PLEK2     4.268419   3.486914   3.293703   0.537386   3.706866   0.652470\n",
       "...            ...        ...        ...        ...        ...        ...\n",
       "TNFSF13   6.748283   6.212415   6.257585  10.367265   6.734800   9.328266\n",
       "AKAP2     7.195798   6.236059   6.895361   4.273324   6.506765   3.872548\n",
       "GZMM     11.561187  10.916939  12.031797   9.859231  14.340722  10.027372\n",
       "PROSER2   2.791149   1.533664   1.805223   1.360205   3.899127   1.952887\n",
       "SLC41A3   4.788027   5.854505   5.468683   3.793310   4.700652   4.015085\n",
       "\n",
       "[773 rows x 6 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.get_data()[2] # reference matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>B</th>\n",
       "      <th>CD4T</th>\n",
       "      <th>CD8T</th>\n",
       "      <th>Mon</th>\n",
       "      <th>NK</th>\n",
       "      <th>Neu</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>liver_apap_1</th>\n",
       "      <td>0.019184</td>\n",
       "      <td>-0.031083</td>\n",
       "      <td>-0.027067</td>\n",
       "      <td>0.019458</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.047001</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_apap_2</th>\n",
       "      <td>0.036571</td>\n",
       "      <td>-0.044764</td>\n",
       "      <td>-0.048988</td>\n",
       "      <td>0.008359</td>\n",
       "      <td>-0.000650</td>\n",
       "      <td>0.050079</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_apap_3</th>\n",
       "      <td>0.015900</td>\n",
       "      <td>-0.038260</td>\n",
       "      <td>-0.041739</td>\n",
       "      <td>0.020647</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.054218</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_apap_4</th>\n",
       "      <td>0.038867</td>\n",
       "      <td>-0.050386</td>\n",
       "      <td>-0.027481</td>\n",
       "      <td>0.022532</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.042313</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_apap_5</th>\n",
       "      <td>0.017044</td>\n",
       "      <td>-0.020135</td>\n",
       "      <td>-0.025081</td>\n",
       "      <td>0.020389</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.027136</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_ctrl_1</th>\n",
       "      <td>0.048748</td>\n",
       "      <td>-0.020959</td>\n",
       "      <td>-0.001581</td>\n",
       "      <td>0.008420</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_ctrl_2</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>-0.011308</td>\n",
       "      <td>0.008132</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.010199</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_ctrl_3</th>\n",
       "      <td>0.030111</td>\n",
       "      <td>-0.015191</td>\n",
       "      <td>-0.011231</td>\n",
       "      <td>0.008002</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_ctrl_4</th>\n",
       "      <td>0.013782</td>\n",
       "      <td>-0.001837</td>\n",
       "      <td>-0.005289</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_ctrl_5</th>\n",
       "      <td>0.017099</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.004770</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_mda_1</th>\n",
       "      <td>0.020509</td>\n",
       "      <td>-0.025078</td>\n",
       "      <td>-0.024952</td>\n",
       "      <td>0.035229</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_mda_2</th>\n",
       "      <td>0.025608</td>\n",
       "      <td>-0.018970</td>\n",
       "      <td>-0.019358</td>\n",
       "      <td>0.026163</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_mda_3</th>\n",
       "      <td>0.025335</td>\n",
       "      <td>-0.020152</td>\n",
       "      <td>-0.019368</td>\n",
       "      <td>0.030794</td>\n",
       "      <td>-0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_mda_4</th>\n",
       "      <td>0.033259</td>\n",
       "      <td>-0.022894</td>\n",
       "      <td>-0.020649</td>\n",
       "      <td>0.028256</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>liver_mda_5</th>\n",
       "      <td>0.033450</td>\n",
       "      <td>-0.025418</td>\n",
       "      <td>-0.025380</td>\n",
       "      <td>0.032199</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen_1</th>\n",
       "      <td>0.139260</td>\n",
       "      <td>0.068860</td>\n",
       "      <td>0.056152</td>\n",
       "      <td>0.087181</td>\n",
       "      <td>0.069142</td>\n",
       "      <td>0.040959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen_2</th>\n",
       "      <td>0.150990</td>\n",
       "      <td>0.065083</td>\n",
       "      <td>0.068107</td>\n",
       "      <td>0.065932</td>\n",
       "      <td>0.079111</td>\n",
       "      <td>0.047688</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen_3</th>\n",
       "      <td>0.159318</td>\n",
       "      <td>0.045537</td>\n",
       "      <td>0.057510</td>\n",
       "      <td>0.081459</td>\n",
       "      <td>0.083528</td>\n",
       "      <td>0.028487</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen_4</th>\n",
       "      <td>0.144145</td>\n",
       "      <td>0.054262</td>\n",
       "      <td>0.071449</td>\n",
       "      <td>0.095491</td>\n",
       "      <td>0.088335</td>\n",
       "      <td>0.011901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>spleen_5</th>\n",
       "      <td>0.139028</td>\n",
       "      <td>0.084378</td>\n",
       "      <td>0.093630</td>\n",
       "      <td>0.069229</td>\n",
       "      <td>0.040382</td>\n",
       "      <td>0.027567</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wbc_1</th>\n",
       "      <td>0.058583</td>\n",
       "      <td>0.065676</td>\n",
       "      <td>0.045102</td>\n",
       "      <td>0.129029</td>\n",
       "      <td>0.003039</td>\n",
       "      <td>0.125378</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wbc_2</th>\n",
       "      <td>0.079303</td>\n",
       "      <td>0.052204</td>\n",
       "      <td>0.028621</td>\n",
       "      <td>0.122273</td>\n",
       "      <td>0.013967</td>\n",
       "      <td>0.135677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wbc_3</th>\n",
       "      <td>0.059922</td>\n",
       "      <td>0.031810</td>\n",
       "      <td>0.009840</td>\n",
       "      <td>0.129348</td>\n",
       "      <td>0.012388</td>\n",
       "      <td>0.156926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wbc_4</th>\n",
       "      <td>0.055744</td>\n",
       "      <td>0.039451</td>\n",
       "      <td>0.033540</td>\n",
       "      <td>0.126413</td>\n",
       "      <td>0.019302</td>\n",
       "      <td>0.140681</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wbc_5</th>\n",
       "      <td>0.087433</td>\n",
       "      <td>0.048113</td>\n",
       "      <td>0.036040</td>\n",
       "      <td>0.130218</td>\n",
       "      <td>0.015967</td>\n",
       "      <td>0.126179</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     B      CD4T      CD8T       Mon        NK       Neu\n",
       "liver_apap_1  0.019184 -0.031083 -0.027067  0.019458 -0.000000  0.047001\n",
       "liver_apap_2  0.036571 -0.044764 -0.048988  0.008359 -0.000650  0.050079\n",
       "liver_apap_3  0.015900 -0.038260 -0.041739  0.020647 -0.000000  0.054218\n",
       "liver_apap_4  0.038867 -0.050386 -0.027481  0.022532  0.000000  0.042313\n",
       "liver_apap_5  0.017044 -0.020135 -0.025081  0.020389  0.000000  0.027136\n",
       "liver_ctrl_1  0.048748 -0.020959 -0.001581  0.008420  0.000000  0.000000\n",
       "liver_ctrl_2  0.000000 -0.000000 -0.011308  0.008132  0.000000  0.010199\n",
       "liver_ctrl_3  0.030111 -0.015191 -0.011231  0.008002 -0.000000  0.000000\n",
       "liver_ctrl_4  0.013782 -0.001837 -0.005289  0.000000  0.000000  0.000000\n",
       "liver_ctrl_5  0.017099 -0.000000 -0.000000  0.004770  0.000000  0.000000\n",
       "liver_mda_1   0.020509 -0.025078 -0.024952  0.035229 -0.000000  0.000000\n",
       "liver_mda_2   0.025608 -0.018970 -0.019358  0.026163  0.000000  0.000000\n",
       "liver_mda_3   0.025335 -0.020152 -0.019368  0.030794 -0.000000  0.000000\n",
       "liver_mda_4   0.033259 -0.022894 -0.020649  0.028256  0.000000  0.000000\n",
       "liver_mda_5   0.033450 -0.025418 -0.025380  0.032199  0.000000  0.000000\n",
       "spleen_1      0.139260  0.068860  0.056152  0.087181  0.069142  0.040959\n",
       "spleen_2      0.150990  0.065083  0.068107  0.065932  0.079111  0.047688\n",
       "spleen_3      0.159318  0.045537  0.057510  0.081459  0.083528  0.028487\n",
       "spleen_4      0.144145  0.054262  0.071449  0.095491  0.088335  0.011901\n",
       "spleen_5      0.139028  0.084378  0.093630  0.069229  0.040382  0.027567\n",
       "wbc_1         0.058583  0.065676  0.045102  0.129029  0.003039  0.125378\n",
       "wbc_2         0.079303  0.052204  0.028621  0.122273  0.013967  0.135677\n",
       "wbc_3         0.059922  0.031810  0.009840  0.129348  0.012388  0.156926\n",
       "wbc_4         0.055744  0.039451  0.033540  0.126413  0.019302  0.140681\n",
       "wbc_5         0.087433  0.048113  0.036040  0.130218  0.015967  0.126179"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dat.get_res() # deconvolution result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "interpreter": {
   "hash": "ed22c18b52bafff8dc0e2325f068b57b9971a18ccb8ccf07b4f22c881d0637c8"
  },
  "kernelspec": {
   "display_name": "Python 3.9.7 ('deconv')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
